substrate/frame/ranked-collective/src/lib.rs

```diff
@@ -763,6 +771,52 @@ pub mod pallet {
 			use frame_support::traits::CallerTrait;
 			o.as_signed().and_then(Self::rank_of)
 		}
+
+		/// Ensure the correctness of the state of this pallet.
+		#[cfg(any(feature = "try-runtime", test))]
+		pub fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+			Self::try_state_members()?;
+			Self::try_state_index()?;
+
+			Ok(())
+		}
+
+		/// ### Invariants of Member storage items
+		///
+		/// Total number of [`Members`] in storage should be >= [`MemberIndex`] of a [`Rank`] in
+		///    [`MemberCount`].
+		#[cfg(any(feature = "try-runtime", test))]
+		fn try_state_members() -> Result<(), sp_runtime::TryRuntimeError> {
+			MemberCount::<T, I>::iter().try_for_each(|(_rank, member_index)| -> DispatchResult {
+				let total_members = Members::<T, I>::iter().count();
+				ensure!(
+				total_members as u32 >= member_index,
+				"Total count of `Members` should be greater than or equal to the number of `MemberIndex` of a particular `Rank` in `MemberCount`."
+				);
+
+				Ok(())
+			})?;
+
+			Ok(())
+		}
```

Comment:
What about checking the rank in `MemberCount` matches the actual member rank?

Reply:
I have added an invariant for this